### PR TITLE
File reorg

### LINKED ISSUE #623: Better align test and R files
Including putting backend variations into their own files.

### LINKED ISSUE #675: Oracle methods should be documented in own topic
Like for SQL server

### PR DESCRIPTION
**Pull request by hadley**
Fixes #623. Fixes #675.

@detule FYI, no need to review (since I think it's basically impossible), but this will hopefully make development a bit easier going forward.

### CODE REVIEW

File: R/driver-terdata.R
Comment (simonpcouch) [inline R/driver-terdata.R:1]: This file should be called "teradata" rather than "terdata"

File: R/odbc-data-type.R [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    character = "VARCHAR(255)",
+    logical = "BIT", # only valid if DB supports Null fields
+    list = "VARCHAR(255)",
+    stop("Unsupported type", call. = FALSE)
+  )
+}
+
+
+#' @export
+`odbcDataType.Redshift` <- function(con, obj, ...) {
```
Comment (simonpcouch) [inline R/odbc-data-type.R:75]: Other databases have their `odbcDataType` method defined in their own file. Any reason why this one is here?

File: R/odbc.R [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
+#' @noMd
+#' @examples
+#' \dontrun{
+#' # Can use spaces or underscores in between words.
+#' odbcSetTransactionIsolationLevel(con, "read uncommitted")
+#'
+#' # Can also use the full constant name.
+#' odbcSetTransactionIsolationLevel(con, "SQL_TXN_READ_UNCOMMITTED")
+#' }
+odbcSetTransactionIsolationLevel <- function(conn, levels) {
```
Comment (simonpcouch) [inline R/odbc.R:16]: This seems like unexpected content for the `odbc.R` file.

File: _pkgdown.yml [hunk 3]
```diff
@@ -21,6 +21,7 @@ reference:
   contents:
   - databricks
   - 'Microsoft SQL Server-class'
+  - 'Oracle-class'
```
Comment (simonpcouch) [inline _pkgdown.yml:24]: I believe this PR also closes #675.

**Review by simonpcouch — APPROVED** [review]
Looking sharp.🙂

File: R/odbc.R (hunk 2 above)
Comment (hadley) [reply R/odbc.R:16]: I'm thinking we'll accumulate random `odbc...` functions in this file.

File: _pkgdown.yml (hunk 3 above)
Comment (hadley) [reply _pkgdown.yml:24]: Good catch, thanks!

**Pull request merged**